### DISCUSSION TITLE
adventures-in-linking/

### DISCUSSION THREAD

**Discussion by giscus** [zmike/zmike.github.io#2] [Announcements]
# adventures-in-linking/

First

https://www.supergoodcode.com/adventures-in-linking/

**Comment by zmike** [comment]
first

**Reply by rhysperry111** [reply]
hey that's cheating >:(

**Reply by qwertychouskie** [reply]
No fair!  I want a refund >:(

**Comment by jsimmons** [comment]
cool post

**Comment by NHOrus** [comment]
Tastes like... pain and interesting programming challenge for someone else to hit their head upon.

**Comment by atlasvirtualreality** [comment]
So mesa never had the code to link this extension to the driver, if so what is to use of it, it seems it's gonna need a good modification of each driver's base code to implement this function, or did you do it already for the upcoming mesa drivers ?

**Reply by zmike** [reply]
There's no testing for this yet, but fixing it is hard so it may take a while.

**Comment by ermarch** [comment]
Two things that comes to mind:
Is it possible to compile the fragment shader before the vertex shader (to get prior information)?

If not it should be possible to reserve the maximum amount of 'slots' based on the largest layout parameter and don't optimize out any. So if the shader defines layout(location = 0) and layout(location = 9) you would actually reserve 10 slots of the largest size, wasting some memory but keeping a working index. I would think.

That said, it might sound silly as I don't have any experience with implementing any of this.

**Reply by zmike** [reply]
The issue is the shaders are compiled in complete isolation, so there's no way to infer anything between stages.
